source/adapters/level_zero/v2/kernel.cpp

```diff
@@ -196,6 +196,48 @@ ur_program_handle_t ur_kernel_handle_t_::getProgramHandle() const {
   return hProgram;
 }
 
+ur_result_t ur_kernel_handle_t_::setExecInfo(ur_kernel_exec_info_t propName,
+                                             const void *pPropValue) {
+  std::scoped_lock<ur_shared_mutex> Guard(Mutex);
+
+  for (auto &kernel : deviceKernels) {
```

Comment:
Yes, it's pretty much a copy-paste. Do you think it would be better to extract this logic from the legacy adapter to some common place and use it here? I didn't do this since the logic is quite simple but there is similar problem with urKernelGetGroupInfo

Reply:
I will share the logic in a separate PR.

Reply:
https://github.com/oneapi-src/unified-runtime/pull/2145